### PR TITLE
add in container vendor

### PR DESCRIPTION
**Pull request by SamuelStuchly**
**What this PR does / why we need it**:
Enables vendoring in container. Adds option to run make vendor in container like other targets, to avoid issues with go module caching depending on what you already have on your machine.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Review by elmiko — APPROVED** [review]
/approve

File: hack/go-mod.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+go mod tidy
+go mod vendor
+go mod verify
```
Comment (JoelSpeed) [inline hack/go-mod.sh:4]: Nit: Should have a new line at the end of this file

**Pull request merged**